code/modules/mob/living/carbon/alien/humanoid/caste/drone.dm

```diff
@@ -43,7 +43,14 @@
 			if(do_after(src, src, 50))
 				var/mob/living/carbon/alien/humanoid/queen/new_xeno = new(loc)
 				mind.transfer_to(new_xeno)
+				var/list/implants = list() //Try to preserve implants.

```

Comment:
This really is the final nitpick I have here

Remove this line and the 2 below it you copypasted everywhere. The only reason they were used originally was for the for(thing in contents-implants) loop. But they don't serve any purpose, just doing the original loop of implants in src is enough.